Anagram.java

```diff
@@ -23,27 +24,84 @@ public static void main(String args[]) {
 			pass = pass && isAnagram(str, randomAnagram);
 			if (!pass) break;
 		}
-		System.out.println(pass ? "test passed" : "test Failed");
-	}  
+		System.out.println(pass ? "test passed" : "test Failed");  
+ 	}  
 
 	// Returns true if the two given strings are anagrams, false otherwise.
-	public static boolean isAnagram(String str1, String str2) {
-		// Replace the following statement with your code
-		return false;
+	public static boolean isAnagram(String str1, String str2) 
+	{
+		str1 = preProcess(str1);
+		str2 = preProcess(str2);
+		String strCheck1 =RemoveSpace(str1);
+		String strCheck2=RemoveSpace(str2);
+
+		if(strCheck1.length() != strCheck2.length())
+			{
+				return false;
+			}
+			for (int i = 0; i < strCheck1.length(); i++) {
+				int cnt1 = 0;
+				int cnt2 = 0;
+				char ch = strCheck1.charAt(i);
+				for (int j = 0; j < strCheck2.length(); j++) {
+
+					if(ch == strCheck2.charAt(j) )
+					{
+						cnt1 ++;
+					}
+					if(ch == strCheck1.charAt(j))
+					{
+						cnt2 ++;
+					}
+				}
+				if(cnt1 != cnt2)
+					return false;
+				
```

Comment:
why nested loop. 
since length are the same we can just go over one string in loop with `indexOf` (if returns -1) -> false, and `substring` to remove the chars.. 
but let go with your code idea (which is correct) 
You are repeating checks 
example:

"aac", "caa"
i = 0, 'a' checked, after inner for, cnt1 = 2, cnt2 =2 
i = 1, 'a' checked (again), after inner for, cnt1 = 2, cnt2 =2 
i = 2, 'c' checked, after inner for, cnt1 = 1, cnt2 = 1  
consider the following: 
```
... // same start of code
int cnt1 = 0;
int cnt2 = 0;
for (int i = 0; i < strCheck1.length(); i++) {
   	char ch = strCheck1.charAt(i);
   	boolean isFirstIndexOfCurrentChar = strCheck1.indexOf(ch) == i; 
   	if (!isFirstIndexOfCurrentChar) {
            continue; 
        }
       // rest of your code; 
}
		
```